### PR TITLE
refactored netlink based ip_notifier in a separated class

### PR DESCRIPTION
**Pull request by aldenml**
This is essentially a copy/paste with the `std::move(cb)` and `const& cb` observations

### CODE REVIEW

File: src/ip_notifier.cpp [hunk 1]
```diff
@@ -339,6 +379,7 @@ struct ip_change_notifier_impl final : ip_change_notifier
 		TORRENT_UNUSED(&ip_change_notifier_impl::on_notify);
 		TORRENT_UNUSED(cb);
 #elif TORRENT_USE_NETLINK
+		using namespace std::placeholders;
```
Comment (arvidn) [inline src/ip_notifier.cpp]: all mentions of ``TORRENT_USE_NETLINK`` and associated snippets of code can be removed now, right?
this function and the constructor specifically

**Comment by arvidn** [comment]
lgtm, but I think some old netlink stuff could be cleaned up

File: src/ip_notifier.cpp (hunk 1 above)
Comment (aldenml) [inline src/ip_notifier.cpp]: yes, but I want to do that at end, when I moved/refactored the windows impl, then delete the old `ip_change_notifier_impl`

**Comment by aldenml** [comment]
Are you OK with the cleanup at the end, I need like two more PRs

**Comment by arvidn** [comment]
sure

**Pull request merged**